services/order.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+from django.db import transaction
+from db.models import Order, Ticket
+
+
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: Optional[str] = None) -> QuerySet:
+    with transaction.atomic():

```

Comment:
Use `transaction.atomic` as decorator instead of context manager. It makes you code cleaner